server/influx_test.go

```diff
@@ -217,6 +217,104 @@ func TestService_Influx_UseCommand(t *testing.T) {
 	}
 }
 
+// TestService_Influx_CommandWithOnClause tests preprocessing of command with ON clause
+func TestService_Influx_CommandWithOnClause(t *testing.T) {
+	tests := []struct {
+		name string
+		db   string
+		rp   string
+	}{
+		{
+			name: "/* no command */",
+		},
+		{
+			name: "SHOW MEASUREMENTS",
+		},
+		{
+			name: "SHOW TAG KEYS ON mydb",
+			db:   "mydb",
+		},
+		{
+			name: "SHOW TAG KEYS ON mydb FROM table",
+			db:   "mydb",
+		},
+		{
+			name: "USE anotherdb; SHOW TAG KEYS ON mydb",
+			db:   "anotherdb",
+		},
+		{
+			name: `show tag keys on "mydb"`,
+			db:   "mydb",
+		},
+		{
+			name: `show tag keys on "mydb" from "table"`,
+			db:   "mydb",
+		},
```

Comment:
Please also add tests for the following edge cases:

```suggestion
		},
		{
			name: `show tag keys on "my_db" from "table"`,
			db:   "my_db",
		},
		{
			name: `show tag keys on "my-db" from "table"`,
			db:   "my-db",
		},
		{
			name: `show tag keys on "my/db" from "table"`,
			db:   "my/db",
		},
		{
			name: `show tag keys on "my db" from "table"`,
			db:   "my db",
		},
```